MLIR.cmake

```diff
@@ -52,300 +55,59 @@ message(STATUS "LLVM_PROJECT_LIB        : " ${LLVM_PROJECT_LIB})
 if (ENV{LLVM_PROJ_BIN})
   set(LLVM_PROJ_BIN $ENV{LLVM_PROJ_BIN})
 else()
-  if(MSVC)
-    if (CMAKE_BUILD_TYPE)
-      set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/${CMAKE_BUILD_TYPE}/bin)
-    else()
-      set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/Release/bin)
-    endif()
-  else()
-    set(LLVM_PROJ_BIN ${LLVM_PROJ_BUILD}/bin)
-  endif()
+  set(LLVM_PROJ_BIN ${LLVM_TOOLS_BINARY_DIR})
 endif()
 message(STATUS "LLVM_PROJ_BIN           : " ${LLVM_PROJ_BIN})
 
-# Include paths for MLIR
-if (USE_INSTALLED_LLVM)
-  set(LLVM_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/include)
-else()
-  set(LLVM_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/llvm/include)
-endif()
-set(LLVM_BIN_INCLUDE_PATH ${LLVM_PROJ_BUILD}/include)
-set(MLIR_SRC_INCLUDE_PATH ${LLVM_PROJ_SRC}/mlir/include)
-set(MLIR_BIN_INCLUDE_PATH ${LLVM_PROJ_BUILD}/tools/mlir/include)
-set(MLIR_TOOLS_DIR ${LLVM_PROJ_BIN})
+# FIXME: Remove after migration
+set(MLIR_TOOLS_DIR ${LLVM_TOOLS_BINARY_DIR})
 
 # ONNX-MLIR tools folder
-if(MSVC)
-  if (CMAKE_BUILD_TYPE)
-    set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin/${CMAKE_BUILD_TYPE})
-  else()
-    set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin/Release)
-  endif()
-else()
-  set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin)
-endif()
+set(ONNX_MLIR_TOOLS_DIR ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/bin)
 message(STATUS "ONNX_MLIR_TOOLS_DIR     : " ${ONNX_MLIR_TOOLS_DIR})
+# Set output library path
+set(ONNX_MLIR_LIB_DIR ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_CFG_INTDIR})
+
 set(ONNX_MLIR_LIT_TEST_SRC_DIR ${ONNX_MLIR_SRC_ROOT}/test/mlir)
 set(ONNX_MLIR_LIT_TEST_BUILD_DIR ${CMAKE_BINARY_DIR}/test/mlir)
 
-set(
-        MLIR_INCLUDE_PATHS
-        ${LLVM_SRC_INCLUDE_PATH};${LLVM_BIN_INCLUDE_PATH};${MLIR_SRC_INCLUDE_PATH};${MLIR_BIN_INCLUDE_PATH}
-)
-include_directories(${MLIR_INCLUDE_PATHS})
+include_directories(${LLVM_INCLUDE_DIRS};${MLIR_INCLUDE_DIRS})
 
-if (NOT USE_INSTALLED_LLVM)
-  # Force CMAKE_INSTALL_PREFIX and BUILD_SHARED_LIBS to be the same as LLVM build
+# If we're pointing at an LLVM build, then assume we want to install into the same
+# install location
+if(EXISTS ${LLVM_PROJ_BUILD}/CMakeCache.txt)
```

Comment:
If CMAKE_INSTALL_PREFIX was explicitly specified, I don't think it should be overwritten.

Reply:
We currently have a dependency that llvm and onnx-mlir bins either both stay in their respective build directory, or both get installed to the same system wide directory (typically /usr/local/bin). See the comments above `getToolPath` in `src/MainUtils.cpp`.

Reply:
> We currently have a dependency that llvm and onnx-mlir bins either both stay in their respective build directory, or both get installed to the same system wide directory (typically /usr/local/bin). See the comments above `getToolPath` in `src/MainUtils.cpp`.

With a complete switch to using MLIR as a package, I don't think there's a need for this requirement because the build of onnx-mlir can be pointed to an installation of LLVM and the paths (like kOptPath) can point to the correct location for the tools. Having said that, if you want to enforce that they be installed together regardless of whether they have to be, then it can stay that way.

Reply:
I'm just pointing out that the current logic of `getToolPath` depends on the assumption made above. If you remove that assumption, the logic of `getToolPath` needs to change accordingly.